api/caasprovisioner/client.go

```diff
@@ -25,14 +26,37 @@ func NewClient(caller base.APICaller) *Client {
 
 // WatchApplications returns a StringsWatcher that notifies of
 // changes to the lifecycles of CAAS applications in the current model.
-func (st *Client) WatchApplications() (watcher.StringsWatcher, error) {
+func (c *Client) WatchApplications() (watcher.StringsWatcher, error) {
 	var result params.StringsWatchResult
-	if err := st.facade.FacadeCall("WatchApplications", nil, &result); err != nil {
+	if err := c.facade.FacadeCall("WatchApplications", nil, &result); err != nil {
 		return nil, err
 	}
 	if err := result.Error; err != nil {
 		return nil, result.Error
 	}
-	w := apiwatcher.NewStringsWatcher(st.facade.RawAPICaller(), result)
+	w := apiwatcher.NewStringsWatcher(c.facade.RawAPICaller(), result)
 	return w, nil
 }
+
+// ApplicationPassword holds parameters for setting
+// an application password.
+type ApplicationPassword struct {
+	Name     string
+	Password string
+}
+
+// SetPasswords sets API passwords for the specified applications.
+func (c *Client) SetPasswords(appPasswords []ApplicationPassword) error {
+	var result params.ErrorResults
+	args := params.EntityPasswords{Changes: make([]params.EntityPassword, len(appPasswords))}
+	for i, p := range appPasswords {
+		args.Changes[i] = params.EntityPassword{
+			Tag: names.NewApplicationTag(p.Name).String(), Password: p.Password,
+		}
+	}
+	err := c.facade.FacadeCall("SetPasswords", args, &result)
+	if err != nil {
+		return err
+	}
+	return result.Combine()
```

Comment:
each change is independent, so the errors shouldn't be combined
also, check the number of results is the same as the number of args?

Reply:
fixed